src/libraries/System.Formats.Tar/tests/TarWriter/TarWriter.WriteEntry.Base.cs

```diff
@@ -48,5 +48,80 @@ protected void VerifyGlobalExtendedAttributesEntry(TarEntry entry, Dictionary<st
                 Assert.Equal(value, gea.GlobalExtendedAttributes[key]);
             }
         }
+
+        public static IEnumerable<object[]> WriteIntField_TheoryData()
+        {
+            foreach (TarEntryFormat format in new[] { TarEntryFormat.V7, TarEntryFormat.Ustar, TarEntryFormat.Pax, TarEntryFormat.Gnu })
+            {
+                // Min value.
+                yield return new object[] { format, 0 };
+
+                yield return new object[] { format, 1 };
+                yield return new object[] { format, 42 };
+
+                // Max value octal.
+                yield return new object[] { format, 0x1FFFFF };
+
+                // These values do not fit the octal representation.
+                bool formatIsOctalOnly = format is not TarEntryFormat.Pax and not TarEntryFormat.Gnu;
+                if (!formatIsOctalOnly)
+                {
+                    // Max value property.
+                    yield return new object[] { format, int.MaxValue };
+                }
+
+            }
+        }
+
+        public static IEnumerable<object[]> WriteTimeStampsWithFormats_TheoryData()
+        {
+            foreach (TarEntryFormat entryFormat in new[] { TarEntryFormat.V7, TarEntryFormat.Ustar, TarEntryFormat.Gnu, TarEntryFormat.Pax })
+            {
+                foreach (DateTimeOffset timestamp in GetWriteTimeStamps(entryFormat))
+                {
+                    yield return new object[] { entryFormat, timestamp };
+                }
+            }
+        }
+
+        public static IEnumerable<object[]> WriteTimeStamps_TheoryData()
```

Comment:
Low pri nit, and I know this is slightly beyond your PR, but if you get a chance, can you please rename this method (and all the invocations) to reflect that this is just for PAX? `WriteTimeStamp_Pax_TheoryData` or something like that.